docs/api/js/utils.md

```diff
@@ -1,68 +1,731 @@
+```
+
+#### Magic value
+
+The value returned from `isEIP1271SignatureCorrect` to confirm signature correctness.
+
+```ts
+export const EIP1271_MAGIC_VALUE = '0x1626ba7e';
+```
+
+#### EIP712 transaction type
+
+Constant representing an EIP712 transaction type.
+
+```ts
+export const EIP712_TX_TYPE = 0x71;
+```
+
+#### Priority op transaction on L2
+
+Constant representing a priority transaction operation on L2.
+
+```ts
+export const PRIORITY_OPERATION_L2_TX_TYPE = 0xff;
+```
+
+#### Max bytecode length
+
+Used for ensuring bytecode length is not over the maximum allowed.
+
+```ts
+export const MAX_BYTECODE_LEN_BYTES = ((1 << 16) - 1) * 32;
+```
+
+### Useful addresses
+
+#### ETH token layer 1
 
 ```typescript
-export const ETH_ADDRESS = "0x0000000000000000000000000000000000000000";
+export const ETH_ADDRESS = '0x0000000000000000000000000000000000000000';
 ```
 
-## ABI of zkSync smart contract
+#### ETH token alias on ZkSync Era
 
 ```typescript
-export const ZKSYNC_MAIN_ABI = new utils.Interface(require("../abi/IZkSync.json"));
+export const L2_ETH_TOKEN_ADDRESS = '0x000000000000000000000000000000000000800a';
 ```
 
-## IERC20 interface
+#### Bootloader
+
+```ts
+export const BOOTLOADER_FORMAL_ADDRESS = '0x0000000000000000000000000000000000008001';
+```
 
-Convenient when interacting with native tokens on zkSync.
+#### Contract deployer
 
-```typescript
-export const IERC20 = new utils.Interface(require("../abi/IERC20.json"));
+```ts
+export const CONTRACT_DEPLOYER_ADDRESS = '0x0000000000000000000000000000000000008006';
+```
+
+#### L1 messenger
+
+```ts
+export const L1_MESSENGER_ADDRESS = '0x0000000000000000000000000000000000008008';
 ```
 
-## Encoding paymaster params
+### Gas
 
-Utility method that returns the correctly formed `paymasterParams` object for the common [paymaster flows](../../dev/developer-guides/aa.md#built-in-paymaster-flows).
+#### `DEFAULT_GAS_PER_PUBDATA_LIMIT`
+
+- Use a large amount of gas per pubdata for signing on layer 2.
+- The amount ensures any reasonable limit is accepted. 
+
+:::info
+- The operator is NOT required to use the actual value and can use any value up to that signed by the user.
+:::
 
 ```typescript
-export function getPaymasterParams(paymasterAddress: Address, paymasterInput: PaymasterInput): PaymasterParams
+export const DEFAULT_GAS_PER_PUBDATA_LIMIT = 50000;
 ```
 
-The definition of the `PaymasterInput` can be found [here](./types.md).
+#### `REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT`
 
-## Useful gas constants
-Currently, there is no method to accurately estimate the required `gasPerPubdataLimit`. That's why for now, it is highly recommended to provide the `DEFAULT_GAS_PER_PUBDATA_LIMIT`. Users are not charged more by providing it.
-Later on it will be possible to query the current recommended limit.
+The current required gas per pubdata for L1->L2 transactions. Any amount is possible for L1->L2 transactions because the cost per gas is adjusted.
```

Comment:
Since this gas per pubdata is required, not any amount is possible : ) 

```suggestion
The current required gas per pubdata for L1->L2 transactions. 
```